src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.task.Task;
+
+/**
+ * Represents a list of Tasks.
+ *
+ * @author Joshua Yong
+ */
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+
+    /**
+     * Class constructor.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Overloaded class constructor which initializes the TaskList with existing tasks.
+     *
+     * @param tasks An ArrayList of tasks to be stored in the TaskList.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    public Task getTask(int i) {
+        return tasks.get(i);
+    }
+
+    public void setTaskAsDone(int i) {
+        tasks.get(i).setDone();
+    }
+
+    public void deleteTask(int i) {
```

Comment:
Maybe you can include more descriptive header comments for the public non-setter and non-getter methods?